ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/attestation/AggregatingAttestationPool.java

```diff
@@ -90,6 +94,7 @@ public synchronized void remove(final Attestation attestation) {
       return;
     }
     attestations.remove(attestation);
+    size.decrementAndGet();
```

Comment:
Discussed offline but this will need to be a little smarter - `attestations.remove` will need to return the number of attestations it actually removed because it intelligently handles aggregates.

So if I remove an aggregate that includes validators 1,2,3,4,5 it will remove any single attestations for the same data from those validators or any combination thereof.  But it won’t remove an aggregate with 1,2,3,4,5,6 because 6 is still new info.

Reply:
I've pushed up the changes which should address this as discussed.  I'm not confident that this is quite right yet, and need to write a couple more tests and talk through this.